LW_EquationTest/LinearEquationTest.cs

```diff
@@ -93,5 +94,218 @@ public void LinearEquationTestIndexerException2()
+        {
+            LinearEquation a = new LinearEquation(1, 2, 3, 4);
+            LinearEquation b = new LinearEquation(2, 1, 0, 5, 1);
+            LinearEquation res = a - b;
+            LinearEquation correct = new LinearEquation(-2, 0, 2, -2, 3);
+            bool ans = (res == correct);
+            Assert.IsTrue(ans);
+        }
+
+        [TestMethod]
+        public void LinearEquationTestInitSame()
+        {
+            LinearEquation a = new LinearEquation(true, 3, 5);
+            LinearEquation res = new LinearEquation(5, 5, 5);
+            bool ans = (a[0] == res[0]) &&
+                       (a[1] == res[1]) &&
+                       (a[2] == res[2]);
+            Assert.IsTrue(ans);
+        }
+
+        [TestMethod]
+        public void LinearEquationTestInitRandom()
+        {
+            LinearEquation a = new LinearEquation(true, 3);
+            bool ans = a != null;
+            Assert.IsTrue(ans);
+        }
+        [TestMethod]
+        public void LinearEquationTestOperatorTrue()
+        {
+            LinearEquation a = new LinearEquation(0, 0, 0, 1);
+            bool ans = false;
+            if (a)
+                ans = true;
+            else
+                ans = false;
+
+            Assert.IsFalse(ans);
+        }
+        [TestMethod]
+        public void LinearEquationTestSolve()
+        {
+            LinearEquation a = new LinearEquation(2, 0, 5);
+
+            float ans = 0;
+            a.Solve(out ans);
+            float trueans = (0F - 5F) / 2F;
+            bool t = ans.CompareTo(trueans) == 0;
+            Assert.IsTrue(t);
+        }
+        [TestMethod]
+        public void LinearEquationTestToString()
+        {
+            LinearEquation a = new LinearEquation(1, 2, 3, 4, 5);
+            bool ans = a.ToString() == "1,2,3,4,5";
+            Assert.IsTrue(ans);
+        }
+        [TestMethod]
+        public void LinearEquationTestMultiplyByNumber1()
+        {
+            LinearEquation a = new LinearEquation(1, 2, 3);
+            LinearEquation result = a.MultiplyByNumber(2);
+            LinearEquation correct = new LinearEquation(2, 4, 6);
+            bool b = result.Equals(correct);
+
+            Assert.IsTrue(b);
+        }
+
+        [TestMethod]
+        public void LinearEquationTestMultiplyByNumber2()
+        {
+            LinearEquation a = new LinearEquation(2, 2, 4);
+            LinearEquation result = a.MultiplyByNumber(1.5F);
+            LinearEquation correct = new LinearEquation(3, 3, 6);
+            bool b = result.Equals(correct);
+
+            Assert.IsTrue(b);
+        }
+        [TestMethod]
+        public void LinearEquationTestOpMinus()
+        {
+            LinearEquation a = new LinearEquation(2, -2, 4);
+            LinearEquation result = -a;
+            LinearEquation correct = new LinearEquation(-2, 2, -4);
+            bool b = result.Equals(correct);
+
+            Assert.IsTrue(b);
```

Comment:
Метод equals не делает то что нужно. Чтобы проверить, что списки содержат одинаковые значения, нужно использовать либо метод SequenceEqual из System.Linq, либо Использовать другой класс 
```suggestion
            bool b = result.SequenceEqual(correct);

            Assert.IsTrue(b);
```
либо
```suggestion
            CollectionAssert.AreEqual(correct,result);
```

Reply:
Комментарий оказался ошибочным, я забыл, что equals реализован 😅